File: service/src/main/java/com/bnorm/barkeep/service/BookService.java
// Copyright 2016 (C) BNORM Software. All rights reserved.
package com.bnorm.barkeep.service;

import java.util.Collection;

import com.bnorm.barkeep.model.Book;

public interface BookService {

  Collection<Book> listBooks();

  Book getBook(long bookId);

  Book createBook(Book book);

  Book setBook(long bookId, Book book);

  void deleteBook(long bookId);
}

File: service/src/main/kotlin/com/bnorm/barkeep/service/BookService.kt
// Copyright 2017 (C) BNORM Software. All rights reserved.
package com.bnorm.barkeep.service

import com.bnorm.barkeep.model.Book
import com.bnorm.barkeep.model.BookSpec

interface BookService {

  fun getBooks(): Collection<Book>

  fun getBook(id: Long): Book

  fun createBook(book: BookSpec): Book

  fun addBookRecipe(bookId: Long, recipeId: Long)

  fun removeBookRecipe(bookId: Long, recipeId: Long)

  fun setBook(id: Long, book: BookSpec): Book

  fun deleteBook(id: Long)
}
